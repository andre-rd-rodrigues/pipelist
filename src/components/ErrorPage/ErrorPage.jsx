import React from "react";
import Button from "components/AppButton/AppButton";
import styles from "./errorpage.module.scss";

function ErrorPage() {
  return (
    <div className={styles.errorPage}>
      <img
        src={`${process.env.PUBLIC_URL}/images/error.png`}
        alt="Pipelist - Network error"
      />
      <h2>Something went wrong</h2>
      <p>
        Please refresh the page and if the problem persists, contact our{" "}
        <a
          href="https://support.pipedrive.com/contact-us"
          rel="noreferrer"
          target="_blank"
        >
          support team.
        </a>
      </p>
      <Button
        color="green"
        label="Refresh"
        onClick={() => window.location.reload()}
      />
    </div>
  );
}

export default ErrorPage;
