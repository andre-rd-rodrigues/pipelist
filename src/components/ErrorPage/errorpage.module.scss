@import "styles/mixins";

.errorPage {
  text-align: center;
  color: $dark;
  & img {
    max-width: 20rem;
  }
  & h2 {
    font-size: 1.8rem;
    margin-bottom: 1.5rem;
  }
  & p,
  button {
    font-size: 0.9rem;
  }
  & a {
    color: $blue;
    text-decoration: none;
    &:hover {
      text-decoration: underline;
    }
  }
}
