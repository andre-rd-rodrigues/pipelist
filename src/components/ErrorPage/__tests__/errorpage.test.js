import reactTestRenderer from "react-test-renderer";
import ErrorPage from "../ErrorPage";

describe("Error page", () => {
  it("renders correctly", () => {
    const wrapper = reactTestRenderer.create(<ErrorPage />);

    expect(wrapper).toMatchSnapshot();
  });
});
