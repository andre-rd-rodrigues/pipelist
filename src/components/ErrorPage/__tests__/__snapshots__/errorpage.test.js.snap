// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error page renders correctly 1`] = `
<div
  className="errorPage"
>
  <img
    alt="Pipelist - Network error"
    src="/images/error.png"
  />
  <h2>
    Something went wrong
  </h2>
  <p>
    Please refresh the page and if the problem persists, contact our
     
    <a
      href="https://support.pipedrive.com/contact-us"
      rel="noreferrer"
      target="_blank"
    >
      support team.
    </a>
  </p>
  <button
    className="appButton"
    color="green"
    label="Refresh"
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": undefined,
        "border": false,
        "color": "white",
      }
    }
  >
    Refresh
  </button>
</div>
`;
